blocks_vertical/procedures.js

```diff
@@ -28,6 +28,289 @@ goog.provide('Blockly.Blocks.procedures');
 goog.require('Blockly.Blocks');
 goog.require('Blockly.constants');
 
+// TODO: Create a namespace properly.
+Blockly.ScratchBlocks.ProcedureUtils = {};
+
+/**
+ * Returns the name of the procedure this block calls, or the empty string if
+ * it has not yet been set.
+ * @return {string} Procedure name.
+ * @this Blockly.Block
+ */
+Blockly.ScratchBlocks.ProcedureUtils.getProcCode = function() {
+  return this.procCode_;
+};
+
+/**
+ * Create XML to represent the (non-editable) name and arguments.
```

Comment:
Done, and now there are two sets of functions (callerMutationToDom and definitionMutationToDom) because not all of the blocks know about the argument names and default values.

I'm open to renaming all of the blocks once I finish defining them all and we know what the full set will be.